src/utils/workflow_utils/src/workflow_utils.c

```diff
@@ -2291,36 +2291,42 @@ ADUC_Result workflow_create_from_inline_step(ADUC_WorkflowHandle base, int stepI
     // Keep only file needed by this step entry. Remove the rest.
     JSON_Array* stepFiles = json_object_get_array(stepObject, ADUCITF_FIELDNAME_FILES);
     JSON_Object* baseFiles = json_object_get_object(updateManifestObject, ADUCITF_FIELDNAME_FILES);
-    int fileCount = (int)json_object_get_count(baseFiles);
-    for (int b = fileCount - 1; b >= 0; b--)
-    {
-        const char* baseFileId = json_object_get_name(baseFiles, b);
+
+    size_t fileCount = json_object_get_count(baseFiles);
+    do {
+        fileCount--;
```

Comment:
if fileCount was 0 then in 1st iteration fileCount would be -1. Perhaps change to while(stepFilesCount > 0) { ... } instead of do...while ?